### PR TITLE
Acm library refactor

### PR DESCRIPTION
**Pull request by kevinbentley**
Replace libacm with one that has an open source library.

### CODE REVIEW

**Comment by JeodC** [comment]
This meant to resolve #83 right? Tested?

**Comment by JeodC** [comment]
Windows release tested. Background music (such as title screen) no longer plays.

**Comment by kevinbentley** [comment]
> Windows release tested. Background music (such as title screen) no longer plays.

Thanks, I'm looking into it.

File: lib/streamaudio.h [hunk 1]
```diff
@@ -143,9 +143,15 @@
 #include "ssl_lib.h"
 #include "TaskSystem.h"
 
+extern "C" {
+#include "libacm.h"
+}
```
Comment (GravisZro) [inline lib/streamaudio.h:148]: Headers are supposed to take care of themselves. Only write `#include "libacm.h"` here.
In libacm.h add this before declarations
```
#ifdef __cplusplus
extern "C" {
#endif

```
and then this after the declarations
```
#ifdef __cplusplus
}
#endif

```

**Comment by DanielGibson** [comment]
Oops, as I announced in #83 yesterday, I did basically the same work, see #135

I also fixed that extern "C" issue and sent a pull request to libacm: https://github.com/markokr/libacm/pull/1

File: stream_audio/streamaudio.cpp [hunk 2]
```diff
@@ -837,12 +894,25 @@ void *AudioStream::StreamCallback(int *size) {
...
-    // We have a compressed stream
-    return m_decoder->Read(m_buffer[buf].data, len);
+  
+  int big_endian = 0;
+  #if OUTRAGE_BIG_ENDIAN
+  big_endian = 1;
+  #endif
+  int len_requested = len;
+  //mprintf((0,"Reading stream: len=%d bytes_read=0,len_requested=%d\n",len,len_requested));
+  int bytes_read = acm_read(m_acm,m_buffer[buf].data,len_requested,big_endian,2,1);
```
Comment (DanielGibson) [inline stream_audio/streamaudio.cpp:904]: this might return a value < 0 on error, and 0 on EOF, I think those cases should be handled?

Comment (kevinbentley) [reply stream_audio/streamaudio.cpp:904]: Good call! Thanks!

File: lib/streamaudio.h (hunk 1 above)
Comment (kevinbentley) [reply lib/streamaudio.h:148]: Fair enough. I didn't want to modify the code from the other library so we could drop in a new version.

**Comment by kevinbentley** [comment]
> I also fixed that extern "C" issue and sent a pull request to libacm: [markokr/libacm#1](https://github.com/markokr/libacm/pull/1)

If they take the PR, that's great. If they don't, I'm conflicted about what to do as far as the libacm.h include comment above. On one hand, we only included that header in one place, so I don't think it's too big of a deal. But I agree that it's not the best approach to wrap the include with the extern c stuff.

**Comment by DanielGibson** [comment]
My PR for libacm has just been merged :)

**Pull request closed without merging**